src/Raven.Studio/typescript/components/common/select/Select.tsx

```diff
@@ -46,14 +52,15 @@ export default function Select<
     return <ReactSelect {...props} className="bs5 react-select-container" classNamePrefix="react-select" />;
 }
 
-export function OptionWithIcon(props: OptionProps<SelectOptionWithIcon>) {
-    const { data } = props;
+export function OptionWithIcon(props: OptionProps<SelectOptionWithIcon<any>>) {
+    const { data, isDisabled } = props;
 
     return (
-        <div style={{ cursor: "default" }}>
+        <div className={isDisabled ? "cursor-not-allowed" : "cursor-pointer"}>
             <components.Option {...props}>
                 {data.icon && <Icon icon={data.icon} color={data.iconColor} />}
                 {data.label}
+                {isDisabled && <LicenseRestrictedBadge licenseRequired={data.licenseRequired} />}
```

Comment:
This component should not be responsible for rendering badge with a license. We use it in other places. If you need one for connection strings then create new in EditConnectionStrings.tsx